### PR TITLE
Faster IO with adam

### PR DESCRIPTION
**Pull request by Malinskiy**
- ktor socket implementation has been deprecated
- new implementation uses vertx with netty internally
- File IO is replaced with a custom implementation

### CODE REVIEW

File: cli/src/main/resources/logback.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<!--
```
Comment (Malinskiy) [inline cli/src/main/resources/logback.xml:1]: This prevents debug log output from netty and vertx

File: vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/AdamDeviceProvider.kt [hunk 2]
```diff
@@ -39,26 +43,24 @@ class AdamDeviceProvider(
...
-    private val bootWaitContext =
-        newFixedThreadPoolContext(androidConfiguration.threadingConfiguration.bootWaitingThreads, "BootWaitThreadPool")
-    override val coroutineContext: CoroutineContext
-        get() = bootWaitContext
-
-    private val adbCommunicationContext: CoroutineContext by lazy {
-        newFixedThreadPoolContext(androidConfiguration.threadingConfiguration.adbIoThreads, "AdbIOThreadPool")
-    }
+    override val coroutineContext: CoroutineContext by lazy { newFixedThreadPoolContext(1, "DeviceMonitor") }
+    private val adbCommunicationContext: CoroutineContext by lazy { newFixedThreadPoolContext(androidConfiguration.threadingConfiguration.adbIoThreads, "AdbIOThreadPool") }
```
Comment (tagantroy) [inline vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/AdamDeviceProvider.kt:51]: Is it scalable for big number of devices?

File: vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/AdamAndroidDevice.kt [hunk 3]
```diff
@@ -77,38 +75,18 @@ class AdamAndroidDevice(
...
+            fetchProps()
+            supportedFeatures = client.execute(FetchDeviceFeaturesRequest(adbSerial))
+            logcatManager.subscribe(this@AdamAndroidDevice)
         }
-        supportedFeatures = client.execute(FetchDeviceFeaturesRequest(adbSerial))
     }
 
     private val dispatcher by lazy {
-        newFixedThreadPoolContext(1, "AndroidDevice - execution - $adbSerial")
+        newFixedThreadPoolContext(2, "AndroidDevice - execution - $adbSerial")
```
Comment (tagantroy) [inline vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/AdamAndroidDevice.kt:88]: Why do we need 2 threads per device?

File: vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/AdamDeviceProvider.kt (hunk 2 above)
Comment (tagantroy) [inline vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/AdamDeviceProvider.kt:51]: Do we need this fixedThreadPoolContext? Can we share some common dispatcher and create context on top of it?

**Review by tagantroy — APPROVED** [review]
LGTM

**Pull request merged**